### PR TITLE
Assert that a single scope is passed to `for_scope`

### PR DESCRIPTION
**Pull request by Urgau**
Addresses https://github.com/rust-lang/rust/pull/118518#issuecomment-1903680468

r? @michaelwoerister

### CODE REVIEW

File: compiler/rustc_session/src/session.rs [hunk 1]
```diff
@@ -1523,16 +1523,25 @@ pub trait RemapFileNameExt {
     where
         Self: 'a;
 
-    fn for_scope(&self, sess: &Session, scopes: RemapPathScopeComponents) -> Self::Output<'_>;
+    /// Returns a possibly remapped filename based on the passed scope and remap cli options.
+    ///
+    /// One and only one scope should be passed to this method. For anything related to
+    /// "codegen" see the [RemapFileNameExt::for_codegen] method.
```
Comment (weihanglo) [inline compiler/rustc_session/src/session.rs:1529]: nit

```suggestion
    /// "codegen" see the [`RemapFileNameExt::for_codegen`] method.
```

**Review by weihanglo — APPROVED** [review]

File: compiler/rustc_session/src/session.rs (hunk 1 above)
Comment (weihanglo) [reply compiler/rustc_session/src/session.rs:1529]: Yeah it's just a personal taste of code being rendered with monospace.

File: compiler/rustc_session/src/session.rs [hunk 2]
```diff
@@ -1523,16 +1523,25 @@ pub trait RemapFileNameExt {
...
 }
 
 impl RemapFileNameExt for rustc_span::FileName {
     type Output<'a> = rustc_span::FileNameDisplay<'a>;
 
-    fn for_scope(&self, sess: &Session, scopes: RemapPathScopeComponents) -> Self::Output<'_> {
-        if sess.opts.unstable_opts.remap_path_scope.contains(scopes) {
+    fn for_scope(&self, sess: &Session, scope: RemapPathScopeComponents) -> Self::Output<'_> {
+        assert!(
+            (scope.bits() & (scope.bits() - 1)) == 0,
```
Comment (michaelwoerister) [inline compiler/rustc_session/src/session.rs:1541]: Would you mind expressing this as `scope.bits().count_ones() == 1`, which is easier to read?

**Review by michaelwoerister — COMMENTED** [review]
Thanks for the PR, @Urgau! Looks good to me with the nit addressed.

File: compiler/rustc_session/src/session.rs (hunk 2 above)
Comment (Urgau) [reply compiler/rustc_session/src/session.rs]: Sure, done.

Comment (michaelwoerister) [reply compiler/rustc_session/src/session.rs]: Thanks, @Urgau! I'll approve once CI has passed.

**Pull request merged**